src/main/java/uk/gov/hmcts/darts/datamanagement/service/impl/DataManagementServiceImpl.java

```diff
@@ -66,6 +74,22 @@ public UUID saveBlobData(String containerName, BinaryData binaryData) {
         return uniqueBlobId;
     }
 
+    @Override
+    public UUID saveBlobData(String containerName, InputStream inputStream) {
+        BlobServiceClient serviceClient = blobServiceFactory.getBlobServiceClient(dataManagementConfiguration.getBlobStorageAccountConnectionString());
+        BlobContainerClient containerClient = blobServiceFactory.getBlobContainerClient(containerName, serviceClient);
+
+        var uniqueBlobId = UUID.randomUUID();
+
+        var client = blobServiceFactory.getBlobClient(containerClient, uniqueBlobId);
+
+        var uploadOptions = new BlobParallelUploadOptions(inputStream);
```

Comment:
Not you're concern here but when I was looking at the download operation I noticed that we were simply letting the Azure Runtime exception propagate in the case of a download failure. Should we throw some kind of darts api media exception in these cases? The reason I mention it is that I didnt know if some upstream logic wanted to handle the failure in some way i.e. retry.

Reply:
Yeah I think for any interaction with Azure, as a minumum we should retry at `darts-api` level. Not sure if BlobClient has that functionality built-in, or if we need to implement it ourselves. In general I think we need to better understand the failure modes around Azure. I know some initial failure mode work was done here: https://tools.hmcts.net/confluence/display/DMP/DARTS+Potential+Failure+Points.